proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -253,12 +254,20 @@ public void connect(ServerInfo info, final Callback<Boolean> callback, final boo
     {
         Preconditions.checkNotNull( info, "info" );
 
-        ServerConnectEvent event = new ServerConnectEvent( this, info, reason );
+        connect( ServerConnectRequest.builder().callback( callback ).retry( retry ).reason( reason ).target( info ).build() );
+    }
+
+    public void connect(final ServerConnectRequest request)
```

Comment:
Redundant finals?

Reply:
These need to be marked final because they are referenced in an inner class for ChannelFutureListener.